src/view/main_scene_view.cpp

```diff
@@ -3,18 +3,79 @@
 #include "view/main_scene_view.hpp"
 
 MainSceneView::MainSceneView()
-: camera(glm::radians(45.0f), 800 / 600, .1f, 100.0f)
+: View()
 {
-	//todo temporary below
-	camera.setPosition(glm::vec3(.0f, 3.0f, 10.0f));
+	camera = new AboveCamera(glm::radians(45.0f), 800 / 600, .1f, 100.0f, 0.05f);
```

Comment:
When you are done changing camera to a smart pointer please use std::make_unique instead of new